lib/elements/tab_expand_controller.dart

```diff
@@ -0,0 +1,209 @@
+import 'dart:async';
+import 'dart:html';
+
+import 'package:meta/meta.dart';
+import 'package:split/split.dart';
+
+import 'button.dart';
+import 'counter.dart';
+import 'elements.dart';
+
+enum TabState {
+  closed,
+  ui,
+  docs,
+  console,
+}
+
+/// Manages the bottom-left panel and tabs
+class TabExpandController {
+  final MDCButton uiOutputButton;
+  final MDCButton consoleButton;
+  final MDCButton docsButton;
+  final MDCButton closeButton;
+  final DElement console;
+  final DElement docs;
+  final Counter unreadCounter;
+  final DElement iframe;
+
+  /// The element to give the top half of the split when this panel
+  /// opens
+  final Element topSplit;
+
+  /// The element to give the bottom half of the split
+  final Element bottomSplit;
+
+  final List<StreamSubscription> _subscriptions = [];
```

Comment:
There is an implicit subscription type of `<dynamic>` here. I think these are all `StreamSubscription<MouseEvent>`.